mailpoet/lib/Logging/LogRepository.php

```diff
@@ -65,12 +66,19 @@ public function getLogs(
     return $query->getQuery()->getResult();
   }
 
-  public function purgeOldLogs(int $daysToKeepLogs) {
-    $queryBuilder = $this->entityManager->createQueryBuilder();
-    return $queryBuilder->delete(LogEntity::class, 'l')
-      ->where('l.createdAt < :days')
-      ->setParameter('days', Carbon::now()->subDays($daysToKeepLogs)->toDateTimeString())
-      ->getQuery()->execute();
+  public function purgeOldLogs(int $daysToKeepLogs, int $limit = 1000) {
+    $logsTable = $this->entityManager->getClassMetadata(LogEntity::class)->getTableName();
+    $this->entityManager->getConnection()->executeStatement("
+      DELETE FROM $logsTable
```

Comment:
I'm considering if we should use here a subquery with ordering to avoid an unexpected gap in logs. Did you think about it? Or what is your opinion here?
It's only an idea that I wanted to mention.